vignettes/poisson.Rmd

```diff
@@ -0,0 +1,359 @@
+
+```{r}
+p_zero <- Poisson(lambda = lambda_zero)
+pdf(p_zero, 0:6)
+```
+
+Second, the slope of $`r round(coef(m)[2], digits = 3)`$ can be interpreted
+as an ability elasticity of the number of goals scored. This is because the
+difference of the log-abilities can also be understood as the log of the ability
+ratio. Thus, when the ability ratio increases by $1$ percent, the expected
+number of goals increases approximately by $`r round(coef(m)[2], digits = 3)`$
+percent.
+
+This yields a different predicted Poisson distribution for each team/match
+in the tournament. We can easily set up the vector of all $128$ `Poisson()`
+distribution objects by extracting the vector of all fitted point estimates
+$(\hat \lambda_1, \dots, \hat \lambda_{128})^\top$:
+
+```{r}
+p_reg <- Poisson(lambda = fitted(m))
+head(p_reg)
+```
+
+As an illustration, the following goal distributions could be expected for the
+final that France won 4-2 against Croatia:
+
+```{r}
+tail(FIFA2018, 2)
+p_final <- tail(p_reg, 2)
+p_final
+pdf(p_final, 0:6)
+```
+
+Clearly, France was expected to score more goals than Croatia
+but both teams scored more goals than expected, albeit not unlikely many.
+
+
+## Further details and extensions
+
+Assuming independence of the number of goals scored, we can obtain
+the table of possible match results (after normal time) by multiplying
+the marginal probabilities (again only up to six goals). In R this be done
+easily using the `outer()` function which by default performs a multiplication
+of its arguments.
+
+```{r}
+res <- outer(pdf(p_final[1], 0:6), pdf(p_final[2], 0:6))
+round(100 * res, digits = 2)
+```
+
+For example, we can see from this table that the expected probability for
+France winning against Croatia 1-0 is $`r round(100 * res[2, 1], 2)`$ percent
+while the probability that France loses 0-1 is only $`r round(100 * res[1, 2], 2)`$
+percent.
+
+The advantage of France can also be brought out more clearly by aggregating
+the probabilities for winning (lower triangular matrix), a draw (diagonal),
+or losing (upper triangular matrix). In R these can be computed as:
+
+```{r}
+sum(res[lower.tri(res)]) ## France wins
+sum(diag(res))           ## draw
+sum(res[upper.tri(res)]) ## France loses
+```
+
+Note that these probabilities do not sum up to $1$ because we only considered
+up to six goals per team but more goals can actually occur with a small
+probability.
+
+Next, we update the expected frequencies table by averaging across the expectations
+per team/match from the regression model.
+
+```{r}
+expctd <- pdf(p_reg, 0:6)
+head(expctd)
+expctd <- colMeans(expctd)
+cbind("observed" = obsrvd, "expected" = expctd)
+```
+
+As before, observed and expected frequencies are reasonably close, emphasizing that
+the model has a good marginal fit for this data. To bring out the deviations graphically
+we show the frequencies on a square root scale using a so-called _hanging rootogram_
```

Comment:
Why would we want to show frequencies on a square root scale?

Reply:
To approximately stabilize the variances of the discrepancies. Otherwise the plot is dominated by the discrepancies for the large frequencies while you don't see those at the small frequencies. I thought that this went to far, though, and just included the reference to the paper that explains the idea.